Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs

```diff
@@ -0,0 +1,117 @@
+        }
+
+        public static string IsValidStatic(ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return IsValidStatic(Instance,
+                                ref matchedPattern,
+                                ref groups,
+                                ref failureLevel,
+                                ref fingerprint,
+                                ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return ValidatorBase.IsValidDynamic(Instance,
+                                                ref fingerprint,
+                                                ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            // We need uri and neither account nor password, or uri and both account and password.  Use XOR
+            if (!groups.TryGetNonEmptyValue("uri", out string uri) ||
+                (groups.TryGetNonEmptyValue("account", out string account) ^
+                groups.TryGetNonEmptyValue("password", out string password)))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Uri = uri,
+                Account = account,
+                Password = password,
+            }.ToString();
+
+            return nameof(ValidationState.Unknown);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText,
+                                                       ref string message)
+        {
+            // TODO: Create a unit test for this. https://github.com/microsoft/sarif-pattern-matcher/issues/258
+
+            var fingerprint = new Fingerprint(fingerprintText);
+            string uri = fingerprint.Uri;
+            string account = fingerprint.Account;
+            string password = fingerprint.Password;
+
+            // At this point account and password must be either both full or both empty.  Only check one
+            if (string.IsNullOrWhiteSpace(account))
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    HttpResponseMessage response = client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead).GetAwaiter().GetResult();
```

Comment:
>client.GetAsync [](start = 51, length = 15)

add try catch

Reply:
Done

---
In reply to: [586924131](https://github.com/microsoft/sarif-pattern-matcher/pull/238#discussion_r586924131) [](ancestors = 586924131)